bin/run.py

```diff
@@ -20,18 +21,24 @@ def __init__(self, problem, submission, testcase):
         self.name = self.testcase.name
         self.result = None
 
-        tmp_path = (
-            self.problem.tmpdir / 'runs' / self.submission.short_path / self.testcase.short_path
+        self.tmpdir = (
+            self.problem.tmpdir
+            / 'runs'
+            / self.submission.short_path
+            / self.testcase.short_path.with_suffix('')
         )
-        self.out_path = tmp_path.with_suffix('.out')
-        self.feedbackdir = tmp_path.with_suffix('.feedbackdir')
+
+        self.in_path = self.tmpdir / 'testcase.in'
+        self.out_path = self.tmpdir / 'testcase.out'
+        self.feedbackdir = self.in_path.with_suffix('.feedbackdir')
```

Comment:
```suggestion
        self.feedbackdir = self.tmpdir / 'testcase.feedbackdir'
```

for consistency?

Reply:
the current version is consistent with the other places where we open the feedbackdir, that's what i would prefer?